File: BankatiProject_Backend/services/User/src/main/java/com/example/user/users/request/UserRequest.java
package com.example.user.users.request;

import com.example.user.users.entity.Role;

import java.util.Date;

public record UserRequest(
         String id,
         String agentId,
         String firstName,
         String lastName,
         String email,
         String address,
         String cin,
         Date birthDate ,
         String phoneNumber,

         Role role,
         String password,
         boolean isFirstLogin ,
         String commercialRn,
         String image,
         String patentNumber ,
         Boolean isPaymentAccountActivated ,
         String typeHissab,
         String currency,
         String token

) {
}

File: BankatiProject_Front-end/src/app/models/UserRequest.ts
export interface UserRequest {
  id: string;
  agentId: string;
  firstName: string;
  lastName: string;
  email: string;
  address: string;
  cin: string;
  birthDate: Date;
  phoneNumber: string;
  role: Role;
  password: string;
  isFirstLogin: boolean;
  commercialRn: string;
  image: string;
  patentNumber: string;
  isPaymentAccountActivated: boolean;
  typeHissab: string;
  currency: string;
  token: string;
}


export enum Role {
  ADMIN = 'ADMIN',
  CLIENT = 'CLIENT',
  AGENT = 'AGENT',
}
